src/sqlfluff/core/config/fluffconfig.py

```diff
@@ -298,20 +299,27 @@ def from_kwargs(
 
         return cls(overrides=overrides, require_dialect=require_dialect)
 
-    def get_templater(
-        self, templater_name: str = "jinja", **kwargs: Any
-    ) -> "RawTemplater":
-        """Fetch a templater by name."""
+    def get_templater_class(self) -> Type["RawTemplater"]:
+        """Get the configured templater class.
+
+        NOTE: This is mostly useful to call directly when rules want to determine
+        the *type* of a templater without (in particular to work out if it's a
+        derivative of the jinja templater), without needing to instantiate a
+        full templater. Instantiated templaters don't pickle well, so aren't
+        automatically passed around between threads/processes.
+        """
         templater_lookup: Dict[str, Type["RawTemplater"]] = {
             templater.name: templater
             for templater in chain.from_iterable(
                 self._plugin_manager.hook.get_templaters()
             )
         }
+        # Fetch the config value.
+        templater_name = self._configs["core"].get("templater", "<no value set>")
         try:
             cls = templater_lookup[templater_name]
             # Instantiate here, optionally with kwargs
-            return cls(**kwargs)
+            return cls
```

Comment:
We aren't actually instantiating the class here any longer. This comment should be updated to reflect that.

Reply:
Ooh good point 👍